web/client/components/maps/MapCard.jsx

```diff
@@ -74,25 +78,40 @@ class MapCard extends React.Component {
     };
 
     render() {
-        var availableAction = [{
-            onClick: (evt) => {this.stopPropagate(evt); this.props.viewerUrl(this.props.map); },
-            glyph: "chevron-right",
-            tooltip: LocaleUtils.getMessageById(this.context.messages, "manager.openInANewTab")
-        }];
-
+        let availableAction = [];
         if (this.props.map.canEdit === true) {
+            availableAction.push(
+                {
+                    onClick: (evt) => {this.stopPropagate(evt); this.displayDeleteDialog(); },
+                    glyph: "trash",
+                    disabled: this.props.map.deleting,
+                    loading: this.props.map.deleting,
+                    tooltip: LocaleUtils.getMessageById(this.context.messages, "manager.deleteMap")
+                }, {
+                    onClick: (evt) => {
+                        this.stopPropagate(evt);
+                        this.onEdit(this.props.map, true);
+                        this.props.detailsSheetActions.onOpenOrFetchDetails({open: false, fetch: true});
+                    },
+                    glyph: "wrench",
+                    disabled: this.props.map.updating,
+                    loading: this.props.map.updating,
+                    tooltip: LocaleUtils.getMessageById(this.context.messages, "manager.editMapMetadata")
+                });
+        }
+        if (this.props.map.details && this.props.map.details !== "NODATA") {
             availableAction.push({
-                onClick: (evt) => {this.stopPropagate(evt); this.onEdit(this.props.map); },
-                glyph: "wrench",
-                disabled: this.props.map.updating,
-                loading: this.props.map.updating,
-                tooltip: LocaleUtils.getMessageById(this.context.messages, "manager.editMapMetadata")
-            }, {
-                onClick: (evt) => {this.stopPropagate(evt); this.displayDeleteDialog(); },
-                glyph: "remove-circle",
-                disabled: this.props.map.deleting,
-                loading: this.props.map.deleting,
-                tooltip: LocaleUtils.getMessageById(this.context.messages, "manager.deleteMap")
+                onClick: (evt) => {
+                    this.stopPropagate(evt);
+                    this.onEdit(this.props.map, false);
+                    if (!this.props.map.detailsText) {
+                        this.props.detailsSheetActions.onOpenOrFetchDetails({open: true, fetch: true, readOnly: true});
```

Comment:
trigger fetch (if details are not present) with an epic removing it from UI part. 
UI should only show the detail text or loading if present but not loaded yet. Pay attention to empty strings